File: src/utils/excalidraw_generator.py
import json

def generate_excalidraw_json(data):
    elements = []
    
    # Canvas base
    canvas_w = data["Size"]["W"]
    canvas_h = data["Size"]["H"]
    
    for zone in data["Zones"]:
        geo = zone["Geometry"]
        is_container = zone["Type"] == "container"
        
        element = {
            "type": "rectangle",
            "x": geo["X"],
            "y": geo["Y"],
            "width": geo["W"],
            "height": geo["H"],
            "strokeColor": "#000000",
            "backgroundColor": "#f3f3f3" if not is_container else "transparent",
            "strokeWidth": 4 if is_container else 1,
            "fillStyle": "solid",
            "roughness": 0,
        }
        elements.append(element)
        
        # Label
        label = {
            "type": "text",
            "x": geo["X"] + 5,
            "y": geo["Y"] + 5,
            "text": f"{zone['ID']}: {zone['Name'] or ''}",
            "fontSize": 16,
            "fontFamily": 1,
            "textAlign": "left"
        }
        elements.append(label)
        
    return {"type": "excalidraw", "elements": elements}

File: src/utils/excalidrawGenerator.ts
import { DashboardLayout, Zone } from './tableauDashboardParser';

export function generateExcalidrawJson(dashboard: DashboardLayout) {
  const elements: any[] = [];
  
  const processZone = (zone: Zone) => {
    const isContainer = zone.type === 'container' || zone.type === 'vert' || zone.type === 'horz';
    
    // Rectangle element
    const rectElement = {
      type: "rectangle",
      x: zone.geometry.x,
      y: zone.geometry.y,
      width: zone.geometry.w,
      height: zone.geometry.h,
      strokeColor: "#000000",
      backgroundColor: isContainer ? "transparent" : "#f3f3f3",
      strokeWidth: isContainer ? 4 : 1,
      fillStyle: "solid",
      roughness: 0,
    };
    elements.push(rectElement);
    
    // Label element
    const labelText = `${zone.id}: ${zone.name || zone.type}`;
    const textElement = {
      type: "text",
      x: zone.geometry.x + 5,
      y: zone.geometry.y + 5,
      text: labelText,
      fontSize: 16,
      fontFamily: 1,
      textAlign: "left"
    };
    elements.push(textElement);

    // Recursively process children
    zone.children.forEach(processZone);
  };

  dashboard.zones.forEach(processZone);

  return {
    type: "excalidraw",
    version: 2,
    source: "Tableau Dashboard Parser",
    elements: elements
  };
}
